### PR TITLE
[web] enable always_specify_types lint

### PR DESCRIPTION
**Pull request by yjbanov**
Specify types everywhere and enable the lint.

### CODE REVIEW

File: lib/web_ui/dev/firefox.dart [hunk 1]
```diff
@@ -56,15 +56,15 @@ class Firefox extends Browser {
   /// [Uri] or a [String].
   factory Firefox(Uri url, {bool debug = false}) {
     final String version = FirefoxArgParser.instance.version;
-    var remoteDebuggerCompleter = Completer<Uri>.sync();
+    Completer<Uri> remoteDebuggerCompleter = Completer<Uri>.sync();
```
Comment (mdebbar) [inline lib/web_ui/dev/firefox.dart:59]:
```suggestion
    final Completer<Uri> remoteDebuggerCompleter = Completer<Uri>.sync();
```

File: lib/web_ui/dev/firefox.dart [hunk 2]
```diff
@@ -84,7 +84,7 @@ user_pref("dom.max_script_run_time", 0);
       File(path.join(temporaryProfileDirectory.path, 'prefs.js'))
           .writeAsStringSync(_profile);
       bool isMac = Platform.isMacOS;
-      var args = [
+      List<String> args = <String>[
```
Comment (mdebbar) [inline lib/web_ui/dev/firefox.dart:87]:
```suggestion
      final List<String> args = <String>[
```

File: lib/web_ui/dev/safari_ios.dart [hunk 3]
```diff
@@ -61,7 +61,7 @@ class SafariIos extends Browser {
       // Uses `xcrun simctl`. It is a command line utility to control the
       // Simulator. For more details on interacting with the simulator:
       // https://developer.apple.com/library/archive/documentation/IDEs/Conceptual/iOS_Simulator_Guide/InteractingwiththeiOSSimulator/InteractingwiththeiOSSimulator.html
-      var process = await io.Process.start('xcrun', [
+      io.Process process = await io.Process.start('xcrun', <String>[
```
Comment (mdebbar) [inline lib/web_ui/dev/safari_ios.dart:64]:
```suggestion
      final io.Process process = await io.Process.start('xcrun', <String>[
```

File: lib/web_ui/dev/safari_macos.dart [hunk 4]
```diff
@@ -64,7 +64,7 @@ class SafariMacOs extends Browser {
       // persistent state and wait until it opens.
       // The details copied from `man open` on macOS.
       // TODO(nurhan): https://github.com/flutter/flutter/issues/50809
-      var process = await Process.start(installation.executable, [
+      Process process = await Process.start(installation.executable, <String>[
```
Comment (mdebbar) [inline lib/web_ui/dev/safari_macos.dart:67]:
```suggestion
      final Process process = await Process.start(installation.executable, <String>[
```

File: lib/web_ui/dev/test_platform.dart [hunk 5]
```diff
@@ -463,21 +463,21 @@ class OneOffHandler {
   ///
   /// [handler] will be unmounted as soon as it receives a request.
   String create(shelf.Handler handler) {
-    var path = _counter.toString();
+    String path = _counter.toString();
```
Comment (mdebbar) [inline lib/web_ui/dev/test_platform.dart:466]:
```suggestion
    final String path = _counter.toString();
```

File: lib/web_ui/dev/test_platform.dart [hunk 6]
```diff
@@ -463,21 +463,21 @@ class OneOffHandler {
...
     _handlers[path] = handler;
     _counter++;
     return path;
   }
 
   /// Dispatches [request] to the appropriate handler.
   FutureOr<shelf.Response> _onRequest(shelf.Request request) {
-    var components = p.url.split(request.url.path);
+    List<String> components = p.url.split(request.url.path);
```
Comment (mdebbar) [inline lib/web_ui/dev/test_platform.dart:474]:
```suggestion
    final List<String> components = p.url.split(request.url.path);
```

File: lib/web_ui/dev/test_platform.dart [hunk 7]
```diff
@@ -463,21 +463,21 @@ class OneOffHandler {
...
     if (components.isEmpty) {
       return shelf.Response.notFound(null);
     }
 
-    var path = components.removeAt(0);
-    var handler = _handlers.remove(path);
+    String path = components.removeAt(0);
```
Comment (mdebbar) [inline lib/web_ui/dev/test_platform.dart:479]:
```suggestion
    final String path = components.removeAt(0);
```

File: lib/web_ui/dev/test_platform.dart [hunk 8]
```diff
@@ -563,9 +563,9 @@ class BrowserManager {
     required PackageConfig packageConfig,
     bool debug = false,
   }) {
-    var browser = _newBrowser(url, browserEnvironment, debug: debug);
+    Browser browser = _newBrowser(url, browserEnvironment, debug: debug);
```
Comment (mdebbar) [inline lib/web_ui/dev/test_platform.dart:566]:
```suggestion
    final Browser browser = _newBrowser(url, browserEnvironment, debug: debug);
```

File: lib/web_ui/dev/test_platform.dart [hunk 9]
```diff
@@ -563,9 +563,9 @@ class BrowserManager {
...
 
-    var completer = Completer<BrowserManager>();
+    Completer<BrowserManager> completer = Completer<BrowserManager>();
```
Comment (mdebbar) [inline lib/web_ui/dev/test_platform.dart:568]:
```suggestion
    final Completer<BrowserManager> completer = Completer<BrowserManager>();
```

File: lib/web_ui/dev/test_platform.dart [hunk 10]
```diff
@@ -612,7 +612,7 @@ class BrowserManager {
     // Start this canceled because we don't want it to start ticking until we
     // get some response from the iframe.
     _timer = RestartableTimer(Duration(seconds: 3), () {
-      for (var controller in _controllers) {
+      for (RunnerSuiteController controller in _controllers) {
```
Comment (mdebbar) [inline lib/web_ui/dev/test_platform.dart:615]:
```suggestion
      for (final RunnerSuiteController controller in _controllers) {
```

File: lib/web_ui/dev/test_platform.dart [hunk 11]
```diff
@@ -658,28 +658,28 @@ class BrowserManager {
       'browser': _browserEnvironment.packageTestRuntime.identifier
     })));
 
-    var suiteID = _suiteID++;
+    int suiteID = _suiteID++;
```
Comment (mdebbar) [inline lib/web_ui/dev/test_platform.dart:661]:
```suggestion
    final int suiteID = _suiteID++;
```

File: lib/web_ui/dev/test_platform.dart [hunk 12]
```diff
@@ -658,28 +658,28 @@ class BrowserManager {
...
+      _channel.sink.add(<String, dynamic>{'command': 'closeSuite', 'id': suiteID});
     }
 
     // The virtual channel will be closed when the suite is closed, in which
     // case we should unload the iframe.
-    var virtualChannel = _channel.virtualChannel();
-    var suiteChannelID = virtualChannel.id;
-    var suiteChannel = virtualChannel.transformStream(
-        StreamTransformer<dynamic, dynamic>.fromHandlers(handleDone: (sink) {
+    VirtualChannel<dynamic> virtualChannel = _channel.virtualChannel();
```
Comment (mdebbar) [inline lib/web_ui/dev/test_platform.dart:673]:
```suggestion
    final VirtualChannel<dynamic> virtualChannel = _channel.virtualChannel();
```

File: lib/web_ui/dev/test_platform.dart [hunk 13]
```diff
@@ -658,28 +658,28 @@ class BrowserManager {
...
+    int suiteChannelID = virtualChannel.id;
```
Comment (mdebbar) [inline lib/web_ui/dev/test_platform.dart:674]:
```suggestion
    final int suiteChannelID = virtualChannel.id;
```

File: lib/web_ui/dev/test_platform.dart [hunk 14]
```diff
@@ -658,28 +658,28 @@ class BrowserManager {
...
+    StreamChannel<dynamic> suiteChannel = virtualChannel.transformStream(
```
Comment (mdebbar) [inline lib/web_ui/dev/test_platform.dart:675]:
```suggestion
    final StreamChannel<dynamic> suiteChannel = virtualChannel.transformStream(
```

File: lib/web_ui/lib/src/engine/platform_dispatcher.dart [hunk 15]
```diff
@@ -715,11 +715,11 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
 
   static List<ui.Locale> parseBrowserLanguages() {
     // TODO(yjbanov): find a solution for IE
-    var languages = html.window.navigator.languages;
+    List<String>? languages = html.window.navigator.languages;
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/platform_dispatcher.dart:718]:
```suggestion
    final List<String>? languages = html.window.navigator.languages;
```

File: lib/web_ui/lib/src/engine/text/ruler.dart [hunk 16]
```diff
@@ -823,7 +823,7 @@ class ParagraphRuler {
     int position = 0;
     final List<html.Node> stack = nodes.reversed.toList();
     while (true) {
-      var node = stack.removeLast();
+      html.Node node = stack.removeLast();
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/text/ruler.dart:826]:
```suggestion
      final html.Node node = stack.removeLast();
```

File: lib/web_ui/test/canvaskit/hot_restart_test.dart [hunk 17]
```diff
@@ -26,7 +26,7 @@ void testMain() {
           assetManager: WebOnlyMockAssetManager());
       expect(windowFlutterCanvasKit, isNotNull);
 
-      var firstCanvasKitInstance = windowFlutterCanvasKit;
+      CanvasKit? firstCanvasKitInstance = windowFlutterCanvasKit;
```
Comment (mdebbar) [inline lib/web_ui/test/canvaskit/hot_restart_test.dart:29]:
```suggestion
      final CanvasKit? firstCanvasKitInstance = windowFlutterCanvasKit;
```

File: lib/web_ui/test/canvaskit/interval_tree_test.dart [hunk 18]
```diff
@@ -15,15 +15,15 @@ void main() {
 void testMain() {
   group('$IntervalTree', () {
     test('is balanced', () {
-      var ranges = <String, List<CodeunitRange>>{
-        'A': [CodeunitRange(0, 5), CodeunitRange(6, 10)],
-        'B': [CodeunitRange(4, 6)],
+      Map<String, List<CodeunitRange>> ranges = <String, List<CodeunitRange>>{
```
Comment (mdebbar) [inline lib/web_ui/test/canvaskit/interval_tree_test.dart:18]:
```suggestion
      final Map<String, List<CodeunitRange>> ranges = <String, List<CodeunitRange>>{
```

File: lib/web_ui/test/canvaskit/interval_tree_test.dart [hunk 19]
```diff
@@ -15,15 +15,15 @@ void main() {
...
+        'A': <CodeunitRange>[CodeunitRange(0, 5), CodeunitRange(6, 10)],
+        'B': <CodeunitRange>[CodeunitRange(4, 6)],
       };
 
       // Should create a balanced 3-node tree with a root with a left and right
       // child.
-      var tree = IntervalTree<String>.createFromRanges(ranges);
-      var root = tree.root;
+      IntervalTree<String> tree = IntervalTree<String>.createFromRanges(ranges);
```
Comment (mdebbar) [inline lib/web_ui/test/canvaskit/interval_tree_test.dart:25]:
```suggestion
      final IntervalTree<String> tree = IntervalTree<String>.createFromRanges(ranges);
```

File: lib/web_ui/test/canvaskit/interval_tree_test.dart [hunk 20]
```diff
@@ -15,15 +15,15 @@ void main() {
...
+      IntervalTreeNode<String> root = tree.root;
```
Comment (mdebbar) [inline lib/web_ui/test/canvaskit/interval_tree_test.dart:26]:
```suggestion
      final IntervalTreeNode<String> root = tree.root;
```

File: lib/web_ui/test/canvaskit/interval_tree_test.dart [hunk 21]
```diff
@@ -32,8 +32,8 @@ void testMain() {
       expect(root.right!.right, isNull);
 
       // Should create a balanced 15-node tree (4 layers deep).
-      var ranges2 = <String, List<CodeunitRange>>{
-        'A': [
+      Map<String, List<CodeunitRange>> ranges2 = <String, List<CodeunitRange>>{
```
Comment (mdebbar) [inline lib/web_ui/test/canvaskit/interval_tree_test.dart:35]:
```suggestion
      final Map<String, List<CodeunitRange>> ranges2 = <String, List<CodeunitRange>>{
```

File: lib/web_ui/test/canvaskit/interval_tree_test.dart [hunk 22]
```diff
@@ -54,8 +54,8 @@ void testMain() {
 
       // Should create a balanced 3-node tree with a root with a left and right
       // child.
-      var tree2 = IntervalTree<String>.createFromRanges(ranges2);
-      var root2 = tree2.root;
+      IntervalTree<String> tree2 = IntervalTree<String>.createFromRanges(ranges2);
```
Comment (mdebbar) [inline lib/web_ui/test/canvaskit/interval_tree_test.dart:57]:
```suggestion
      final IntervalTree<String> tree2 = IntervalTree<String>.createFromRanges(ranges2);
```

File: lib/web_ui/test/canvaskit/interval_tree_test.dart [hunk 23]
```diff
@@ -54,8 +54,8 @@ void testMain() {
...
+      IntervalTreeNode<String> root2 = tree2.root;
```
Comment (mdebbar) [inline lib/web_ui/test/canvaskit/interval_tree_test.dart:58]:
```suggestion
      final IntervalTreeNode<String> root2 = tree2.root;
```

File: lib/web_ui/test/canvaskit/interval_tree_test.dart [hunk 24]
```diff
@@ -68,16 +68,16 @@ void testMain() {
     });
 
     test('finds values whose intervals overlap with a given point', () {
-      var ranges = <String, List<CodeunitRange>>{
-        'A': [CodeunitRange(0, 5), CodeunitRange(7, 10)],
-        'B': [CodeunitRange(4, 6)],
+      Map<String, List<CodeunitRange>> ranges = <String, List<CodeunitRange>>{
```
Comment (mdebbar) [inline lib/web_ui/test/canvaskit/interval_tree_test.dart:71]:
```suggestion
      final Map<String, List<CodeunitRange>> ranges = <String, List<CodeunitRange>>{
```

File: lib/web_ui/test/canvaskit/interval_tree_test.dart [hunk 25]
```diff
@@ -68,16 +68,16 @@ void testMain() {
...
+        'A': <CodeunitRange>[CodeunitRange(0, 5), CodeunitRange(7, 10)],
+        'B': <CodeunitRange>[CodeunitRange(4, 6)],
       };
-      var tree = IntervalTree<String>.createFromRanges(ranges);
+      IntervalTree<String> tree = IntervalTree<String>.createFromRanges(ranges);
```
Comment (mdebbar) [inline lib/web_ui/test/canvaskit/interval_tree_test.dart:75]:
```suggestion
      final IntervalTree<String> tree = IntervalTree<String>.createFromRanges(ranges);
```

File: lib/web_ui/test/golden_tests/engine/canvas_image_blend_mode_test.dart [hunk 26]
```diff
@@ -36,14 +36,19 @@ void testMain() async {
...
 
-  List<List<BlendMode>> modes = [[BlendMode.clear, BlendMode.src, BlendMode.dst,
-    BlendMode.srcOver, BlendMode.dstOver, BlendMode.srcIn, BlendMode.dstIn, BlendMode.srcOut],
-    [BlendMode.dstOut, BlendMode.srcATop, BlendMode.dstATop, BlendMode.xor,
-    BlendMode.plus, BlendMode.modulate, BlendMode.screen, BlendMode.overlay],
-    [BlendMode.darken, BlendMode.lighten, BlendMode.colorDodge, BlendMode.hardLight,
-    BlendMode.softLight, BlendMode.difference, BlendMode.exclusion, BlendMode.multiply],
-    [BlendMode.hue, BlendMode.saturation, BlendMode.color,
-    BlendMode.luminosity]];
+  List<List<BlendMode>> modes = <List<BlendMode>>[
```
Comment (mdebbar) [inline lib/web_ui/test/golden_tests/engine/canvas_image_blend_mode_test.dart:39]:
```suggestion
  final List<List<BlendMode>> modes = <List<BlendMode>>[
```

File: lib/web_ui/test/golden_tests/engine/canvas_stroke_joins_golden_test.dart [hunk 27]
```diff
@@ -48,13 +48,13 @@ void paintStrokeJoins(BitmapCanvas canvas) {
   Offset mid = Offset(120, 10);
   Offset end = Offset(120, 20);
 
-  var strokeCaps = [StrokeCap.butt, StrokeCap.round, StrokeCap.square];
+  List<StrokeCap> strokeCaps = <StrokeCap>[StrokeCap.butt, StrokeCap.round, StrokeCap.square];
```
Comment (mdebbar) [inline lib/web_ui/test/golden_tests/engine/canvas_stroke_joins_golden_test.dart:51]:
```suggestion
  final List<StrokeCap> strokeCaps = <StrokeCap>[StrokeCap.butt, StrokeCap.round, StrokeCap.square];
```

File: lib/web_ui/test/golden_tests/engine/canvas_stroke_joins_golden_test.dart [hunk 28]
```diff
@@ -48,13 +48,13 @@ void paintStrokeJoins(BitmapCanvas canvas) {
...
   for (StrokeCap cap in strokeCaps) {
-    var joints = [StrokeJoin.miter, StrokeJoin.bevel, StrokeJoin.round];
-    var colors = [Color(0xFFF44336), Color(0xFF4CAF50), Color(0xFF2196F3)]; // red, green, blue
+    List<StrokeJoin> joints = <StrokeJoin>[StrokeJoin.miter, StrokeJoin.bevel, StrokeJoin.round];
```
Comment (mdebbar) [inline lib/web_ui/test/golden_tests/engine/canvas_stroke_joins_golden_test.dart:53]:
```suggestion
    final List<StrokeJoin> joints = <StrokeJoin>[StrokeJoin.miter, StrokeJoin.bevel, StrokeJoin.round];
```

File: lib/web_ui/test/golden_tests/engine/canvas_stroke_joins_golden_test.dart [hunk 29]
```diff
@@ -48,13 +48,13 @@ void paintStrokeJoins(BitmapCanvas canvas) {
...
+    List<Color> colors = <Color>[Color(0xFFF44336), Color(0xFF4CAF50), Color(0xFF2196F3)]; // red, green, blue
```
Comment (mdebbar) [inline lib/web_ui/test/golden_tests/engine/canvas_stroke_joins_golden_test.dart:54]:
```suggestion
    final List<Color> colors = <Color>[Color(0xFFF44336), Color(0xFF4CAF50), Color(0xFF2196F3)]; // red, green, blue
```

File: lib/web_ui/test/golden_tests/engine/canvas_stroke_joins_golden_test.dart [hunk 30]
```diff
@@ -48,13 +48,13 @@ void paintStrokeJoins(BitmapCanvas canvas) {
...
     for (int i = 0; i < joints.length; i++) {
-      var join = joints[i];
-      var color = colors[i % colors.length];
+      StrokeJoin join = joints[i];
+      Color color = colors[i % colors.length];
```
Comment (mdebbar) [inline lib/web_ui/test/golden_tests/engine/canvas_stroke_joins_golden_test.dart:57]:
```suggestion
      final StrokeJoin join = joints[i];
      final Color color = colors[i % colors.length];
```

File: lib/web_ui/test/golden_tests/engine/draw_vertices_golden_test.dart [hunk 31]
```diff
@@ -402,8 +402,8 @@ Future<HtmlImage> createTestImage({int width = 50, int height = 40}) {
   ctx.fillRect(2 * width / 3, 0, width / 3, height);
   ctx.fill();
   html.ImageElement imageElement = html.ImageElement();
-  Completer<HtmlImage> completer = Completer();
-  imageElement.onLoad.listen((event) {
+  Completer<HtmlImage> completer = Completer<HtmlImage>();
```
Comment (mdebbar) [inline lib/web_ui/test/golden_tests/engine/draw_vertices_golden_test.dart:405]:
```suggestion
  final Completer<HtmlImage> completer = Completer<HtmlImage>();
```

File: lib/web_ui/test/keyboard_converter_test.dart [hunk 32]
```diff
@@ -51,7 +51,7 @@ void testMain() {
       return key.type == ui.KeyEventType.down;
     });
     bool preventedDefault = false;
-    final onPreventDefault = () { preventedDefault = true; };
+    final Null Function() onPreventDefault = () { preventedDefault = true; };
```
Comment (mdebbar) [inline lib/web_ui/test/keyboard_converter_test.dart:54]: Curious why you didn't use `VoidCallback`?

**Review by mdebbar — APPROVED** [review]

File: lib/web_ui/dev/common.dart [hunk 33]
```diff
@@ -221,7 +221,7 @@ class DevNull implements StringSink {
   void write(Object? obj) {}
 
   @override
-  void writeAll(Iterable objects, [String separator = ""]) {}
+  void writeAll(Iterable<dynamic> objects, [String separator = ""]) {}
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply lib/web_ui/dev/common.dart:224]: The `StringSink` interface spells it out as `Iterable<dynamic>`. I didn't want to change that.

File: lib/web_ui/test/keyboard_converter_test.dart (hunk 32 above)
Comment (yjbanov) [reply lib/web_ui/test/keyboard_converter_test.dart:54]: Done.

**Comment by yjbanov** [comment]
I will address `final` in a separate PR where I'm going to enable `prefer_final_locals`.

**Pull request merged**